### PR TITLE
FORMS-253 # Dont mutate the definition object

### PR DESCRIPTION
**Pull request by simonmarklar**
- Replace  definition mutations in forms with setting of model attributes after initialisation

### CODE REVIEW

**Comment by jokeyrhyme** [comment]
https://github.com/blinkmobile/varied-definition.js/pull/1

File: forms/models/elements/subform.js [hunk 1]
```diff
@@ -195,9 +195,6 @@ define(function (require) {
                 return memo;
               }
 
-              // let the element know who its parent is
-              element.parentElement = self;
```
Comment (jokeyrhyme) [inline forms/models/elements/subform.js:199]: Huh. So we were setting this on both the SubForm and the SubFormElement. Weird.

Is it worth double-checking this with @samblinkmobile to see if any of the code he has written depends on the `SubFormElement#parentElement` property? We don't depend on it internally, it seems, as the tests pass.

**Comment by jokeyrhyme** [comment]
If we hold this until https://github.com/blinkmobile/varied-definition.js/pull/1 is merged and released, then this PR should probably include the update to package.json that refers to the new version, right?

File: forms/models/elements/subform.js (hunk 1 above)
Comment (simonmarklar) [inline forms/models/elements/subform.js:199]: arg, there should be a modification on line 203 to set this on the actual element.

**Comment by simonmarklar** [comment]
yeah, we can do it that way. I just need to know what the version number for varied definition would be.  We're not really breaking compatability but ifyou are expecting the source to be modified then we are, so is it a major or a minute bump?

**Comment by jokeyrhyme** [comment]
LGTM :+1:

**Pull request merged**